dm/dm/worker/source_worker.go

```diff
@@ -249,9 +249,12 @@ func (w *SourceWorker) Stop(graceful bool) {
 // updateSourceStatus updates w.sourceStatus.
 func (w *SourceWorker) updateSourceStatus(ctx context.Context) error {
 	w.sourceDBMu.Lock()
+	w.RLock()
+	cfg := w.cfg
```

Comment:
seems this lock can not avoid data-race? w.cfg is a pointer

Reply:
This lock is used to prevent the situation that `w.cfg` is updated in while we are trying to access `w.cfg` at https://github.com/pingcap/tiflow/blob/f29084c/dm/dm/worker/source_worker.go#L1286

Reply:
ok, `cfg` here points to `oldCfg` and at first, I wondered if there was a problem with correctness, e.g: getting the `cfg` before the update, but there is no problem because the source cfg only  allowed to be updated when there is no running tasking(running source worker)

Reply:
To me there is no big problem. `RefreshSourceCfg` won't revise cfg.SourceID. The other race place may let the user get wrong source IP, but that's not a big deal.